File: common/game-settings.ts
import { AiType } from './ai-name';

export enum StartingPlayer {
    Player1,
    Player2,
}

export enum RoomType {
    public,
    private,
}
export enum NumberOfPlayer {
    OneVone,
    OneVthree,
}
export class GameSettings {
    password: string;
    gameType: NumberOfPlayer;
    constructor(
        public creatorName: string,
        public startingPlayer: StartingPlayer,
        public timeMinute: string,
        public timeSecond: string,
        public level: AiType,
        public dictionary: string,
        public type: RoomType,
        gameType: NumberOfPlayer
    ) {
        this.password = '';
        this.gameType = gameType;
    }
}

File: prototypeLeger/app/src/main/java/com/example/scrabbleprototype/model/GameSettings.kt
package com.example.scrabbleprototype.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class StartingPlayer {
    Player1,
    Player2,
}

@Serializable
enum class AiType {
    beginner,
    expert,
}

@Serializable
enum class RoomType {
    public,
    private,
}

@Serializable
enum class NumberOfPlayer {
    OneVone,
    OneVthree
}

@Serializable
class GameSettings(var creatorName: String, var startingPlayer: StartingPlayer, var timeMinute: String,
                   var timeSecond: String, var level: AiType, var dictionary: String, var type: Int,
                   var gameType: Int) {
    var password: String = ""
}
